### PR TITLE
Add translation to srv metric

### PR DESCRIPTION
**Pull request by Abdellaoui-Souhail**
## Checklist

- [ ] My pull request has a clear and explanatory title.
- [ ] If necessary, my code is [vectorized](https://www.geeksforgeeks.org/vectorization-in-python/).
- [ ] I added appropriate unit tests.
- [ ] I made sure the code passes all unit tests. (refer to comment below)
- [ ] My PR follows [PEP8](https://peps.python.org/pep-0008/) guidelines. (refer to comment below)
- [ ] My PR follows [geomstats coding style](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#coding-style-guidelines) and API.
- [ ] My code is properly documented and I made sure the documentation renders properly. ([Link](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#documentation))
- [ ] I linked to issues and PRs that are relevant to this PR.

### CODE REVIEW

File: geomstats/geometry/discrete_curves.py [hunk 1]
```diff
@@ -1188,10 +1190,141 @@ class SRVMetric(PullbackDiffeoMetric):
...
+        discrete_curves.equip_with_metric(
+            SRVTranslationMetric, fiber_bundle=SRVTranslationBundle
+        )
+
+        factors = [space.ambient_manifold, discrete_curves]
+        embedding_space = ProductManifold(factors=factors, default_point_type="auto")
+        return embedding_space
+
+    def diffeomorphism(self, curves):
+        r"""Compute the image of a curve."""
```
Comment (ninamiolane) [inline geomstats/geometry/discrete_curves.py:1219]: Missing parameters description in docstrings: add them, here and everywhere

File: geomstats/geometry/discrete_curves.py [hunk 2]
```diff
@@ -1188,10 +1190,141 @@ class SRVMetric(PullbackDiffeoMetric):
...
+
+        curves = curves_start_origin + starting_points
+
+        if curve_ndim == 1:
+            return gs.squeeze(curves)
+
+        return curves
+
+    def tangent_diffeomorphism(self, tangent_vec, point):
+        r"""Equivalent to diffeomorphism."""
```
Comment (ninamiolane) [inline geomstats/geometry/discrete_curves.py:1267]: No need to add "r" in front of a docstring if you do not have mathematical equation in it. Remove it.

Also the docstring should start with a verb at infinitive, e.g. "Compute the ..."

--> adapt the docstrings to these conventions

Comment (ninamiolane) [reply geomstats/geometry/discrete_curves.py:1267]: https://geomstats.github.io/contributing/index.html#writing-docstrings

File: geomstats/geometry/discrete_curves.py [hunk 3]
```diff
@@ -1188,10 +1190,141 @@ class SRVMetric(PullbackDiffeoMetric):
...
+        return self.diffeomorphism(tangent_vec)
+
+    def inverse_tangent_diffeomorphism(self, tangent_vec, image_point):
+        r"""Equivalent to inverse diffeomorphism."""
+        return self.inverse_diffeomorphism(tangent_vec)
+
+
+class SRVTranslationBundle(FiberBundle):
+    r"""Bundle of curves when you quotient by translation."""
```
Comment (ninamiolane) [inline geomstats/geometry/discrete_curves.py:1276]: missing parameters of init here.

Comment (ninamiolane) [reply geomstats/geometry/discrete_curves.py:1276]: And in the other python classes as well: please add them everywhere by taking example on other python classes of geomstats

File: tests/data/discrete_curves_data.py [hunk 4]
```diff
@@ -127,6 +129,102 @@ class SRVMetricTestData(_RiemannianMetricTestData):
...
+    n_tangent_vecs_list = random.sample(range(2, 5), 2)
+    n_points_a_list = [1, 2]
+    n_points_b_list = [1, 2]
+    batch_size_list = random.sample(range(2, 5), 2)
+    alpha_list = [1] * 2
+    n_rungs_list = [1] * 2
+    scheme_list = ["pole"] * 2
+
+    Metric = SRVTranslationMetric
+    Fiber_bundle = SRVTranslationBundle
```
Comment (ninamiolane) [inline tests/data/discrete_curves_data.py:226]: no mix of _ and camelcases. Use `Bundle = `

File: tests/tests_geomstats/test_discrete_curves.py [hunk 5]
```diff
@@ -110,9 +112,105 @@ class TestSRVMetric(RiemannianMetricTestCase, metaclass=Parametrizer):
 
     testing_data = SRVMetricTestData()
 
-    def test_srv_transform_and_srv_transform_inverse(self, space, rtol, atol):
+    def test_diffeomorphism_and_inverse_diffeomorphism(self, space, rtol, atol):
+        """Test that srv and its inverse are inverse."""
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_discrete_curves.py:116]: is that docstring still accurate? If not, update.

File: tests/tests_geomstats/test_discrete_curves.py [hunk 6]
```diff
@@ -110,9 +112,105 @@ class TestSRVMetric(RiemannianMetricTestCase, metaclass=Parametrizer):
...
+        space.equip_with_metric(self.Metric)
+        curve = space.random_point(n_samples=2)
+
+        image = space.metric.diffeomorphism(curve)
+        inverse_image = space.metric.inverse_diffeomorphism(image)
+
+        result = inverse_image.shape
+        expected = (curve.shape[0], curve.shape[1], 3)
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_discrete_curves.py:124]: why not writing curve.shape here?

File: tests/tests_geomstats/test_discrete_curves.py [hunk 7]
```diff
@@ -110,9 +112,105 @@ class TestSRVMetric(RiemannianMetricTestCase, metaclass=Parametrizer):
...
+        tangent_vec = space.random_point(n_samples=2)
+
+        image = space.metric.diffeomorphism(curve)
+        image_tangent = space.metric.tangent_diffeomorphism(tangent_vec, curve)
+        inverse_image = space.metric.inverse_tangent_diffeomorphism(
+            image_tangent, image
+        )
+
+        result = inverse_image.shape
+        expected = (tangent_vec.shape[0], tangent_vec.shape[1], 3)
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_discrete_curves.py:146]: ditto

File: tests/tests_geomstats/test_discrete_curves.py [hunk 8]
```diff
@@ -110,9 +112,105 @@ class TestSRVMetric(RiemannianMetricTestCase, metaclass=Parametrizer):
 
     testing_data = SRVMetricTestData()
 
-    def test_srv_transform_and_srv_transform_inverse(self, space, rtol, atol):
+    def test_diffeomorphism_and_inverse_diffeomorphism(self, space, rtol, atol):
+        """Test that srv and its inverse are inverse."""
+        space.equip_with_metric(self.Metric)
+        curve = space.random_point(n_samples=2)
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_discrete_curves.py:118]: don't hardcode n_samples, but use an input to the test function

File: tests/tests_geomstats/test_discrete_curves.py [hunk 9]
```diff
@@ -110,9 +112,105 @@ class TestSRVMetric(RiemannianMetricTestCase, metaclass=Parametrizer):
...
+        result = inverse_image
+        expected = curve
+        self.assertAllClose(result, expected, rtol, atol)
+
+    def test_tangent_diffeomorphism_and_inverse_tangent_diffeomorphism(
+        self, space, rtol, atol
+    ):
         """Test that srv and its inverse are inverse."""
         space.equip_with_metric(self.Metric)
+        curve = space.random_point(n_samples=2)
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_discrete_curves.py:136]: Ditto

**Comment by alebrigant** [comment]
After looking at this PR again, I realize that there are many things that need to be corrected or removed. After discussing with @luisfpereira, we have come to the conclusion that the easiest was for to Luis to start a new PR to fix these issues. @MMalikT it will be easier for you to iterate on the work of Luis, once everything is clean.

**Pull request closed without merging**